### ISSUE TITLE
Confusing error if a user renames a workspace directory.

### ISSUE THREAD

**Issue by nelhage-stripe** [bazelbuild/bazel#2857]
### Description of the problem / feature request / question:

If I rename a directory containing a bazel workspace, future builds fail with an extremely confusing error (at least, for a bazel non-expert).

The failure is because the first run, it creates the `bazel-<name>` symlink, but on the future run in a different directory name, it tries to traverse that symlink and finds all the downloaded `BUILD` files in the `external/` directory.

It'd be nice if this worked, or at least failed with an intelligible error. This caused some confusion in the process of introducing bazel to our team.

### If possible, provide a minimal example to reproduce the problem:
```
$ git clone git@github.com:bazelbuild/buildtools
$ cd buildtools/ && bazel test //...
…
$ cd ..
$ mv buildtools buildifier
$ cd buildifier/ && bazel test //...
.........
ERROR: package contains errors: bazel-buildtools/external/io_bazel/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/resourcejar.
ERROR: error loading package 'bazel-buildtools/external/io_bazel/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/resourcejar': Extension file not found. Unable to load package for '//tools/build_rules:java_rules_skylark.bzl': BUILD file not found on package path.
INFO: Elapsed time: 4.890s
ERROR: Couldn't start the build. Unable to run tests.
$
```

### Environment info
* Operating System:
Observed on both Linux and macOS.

* Bazel version (output of `bazel info release`):
`release 0.4.5`

* If `bazel info release` returns "development version" or "(@non-git)", please tell us what source tree you compiled Bazel from; git commit hash is appreciated (`git rev-parse HEAD`):

###  Have you found anything relevant by searching the web?
(e.g. [StackOverflow answers](http://stackoverflow.com/questions/tagged/bazel),
[GitHub issues](https://github.com/bazelbuild/bazel/issues),
email threads on the [`bazel-discuss`](https://groups.google.com/forum/#!forum/bazel-discuss) Google group)

### Anything else, information or logs or outputs that would be helpful?
(If they are large, please upload as attachment or provide link).

**Comment by aehlig** [comment]
Yes, that's a serious usability issue. Personally, also for this very reason, I have `build --symlink_prefix=/` in my `.bazelrc`.

**Comment by kchodorow** [comment]
This should be fixed by #1681, but leaving this open to make sure there's a test that changing the `workspace()` name doesn't trigger the same issue.

**Comment by dslomov** [comment]
This is now fixed
